### PR TITLE
WINDUPRULE-295: adding JTA technology stats rules

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.jboss.org/browse/WINDUPRULE-295
depends on https://github.com/windup/windup/pull/1179

### CODE REVIEW

File: rules-reviewed/technology-usage/jta-technology-usage.windup.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <targetTechnology id="eap"/>
+        <phase>PostMigrationRulesPhase</phase>
+    </metadata>
+    <rules>
+        <rule id="technology-usage-jta-00010">
+            <when>
+                <graph-query discriminator="DataSourceModel">
```
Comment (mareknovotny) [inline rules-reviewed/technology-usage/jta-technology-usage.windup.xml]: Replacing with JNDIResourceModel (parent data model) doesn't help

**Comment by mrizzi** [comment]
I had to do some ad-hoc changes to [QueryHandler:79](https://github.com/windup/windup/blob/master/config-xml/addon/src/main/java/org/jboss/windup/config/parser/xml/when/QueryHandler.java#L79) to have the "true" comparison to work:
```java           
                if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false"))
                {
                    query.withProperty(propertyName, Boolean.valueOf(value));
                } else
                {
                    query.withProperty(propertyName, value);
                }
```
The issue was that the property from the graph is a Boolean and we compare with the string `true` so it failed.
Another minor issue is that i had to set the attribute `number-found` to 2 to have the test [jta-technology-usage.windup.test.xml](https://github.com/windup/windup-rulesets/pull/270/files#diff-873b674ca67b106c95a1d0eb0d8d2307R10) to work.
Tomorrow i'll implement a better solution using an optional attribute in `<property>` tag to handle the type of the tag's content (default: string; available: string, boolean, ...) and send you a PR.

**Comment by mareknovotny** [comment]
oh, that type conversion made my day, thanks @mrizzi  for discovery.

**Comment by mareknovotny** [comment]
retest

**Pull request merged**